core/store/orm/config.go

```diff
@@ -703,7 +734,11 @@ func (c Config) EthNonceAutoSync() bool {
 
 // EthGasLimitDefault sets the default gas limit for outgoing transactions.
 func (c Config) EthGasLimitDefault() uint64 {
-	return c.getWithFallback("EthGasLimitDefault", parseUint64).(uint64)
+	ethGasLimitDefault := chainSpecificConfig(c).EthGasLimitDefault
+	if c.viper.IsSet(EnvVarName("EthGasLimitDefault")) || ethGasLimitDefault == 0 {
```

Comment:
I think the left side of the conditional will _always_ be true.

Because a default is set in schema.go.

How about changing this:

```
49:	EthGasLimitDefault                        uint64          `env:"ETH_GAS_LIMIT_DEFAULT" default:"500000"`
```

To this:

```
49:	EthGasLimitDefault                        uint64          `env:"ETH_GAS_LIMIT_DEFAULT"`
```

Then setting the default ChainSpecificDefaults to be 500000